tratto/src/main/java/star/tratto/util/javaparser/JavaParserUtils.java

```diff
@@ -471,52 +656,189 @@ public static String getMethodSignature(MethodDeclaration methodDeclaration) {
+        String methodModifiers = methodDeclaration.toString().startsWith("ReflectionMethodDeclaration") ? matcher.group(1) : matcher.group(2);
+        // Take into account the case in which the method declaration refers to a method without an access specifier.
         if (methodModifiers == null) {
-            assert methodDeclaration.startsWith("ReflectionMethodDeclaration");
+            assert methodDeclaration.toString().startsWith("ReflectionMethodDeclaration");
             methodModifiers = "";
         }
-        List<String> methodTypeParameters = new ArrayList<>();
-        for (int i=0; i < methodUsage.getDeclaration().getTypeParameters().size(); i++) {
-            methodTypeParameters.add(methodUsage.getDeclaration().getTypeParameters().get(i).getName());
+        List<String> typeParameterList = getMethodUsageTypeParameters(methodUsage);
+        List<String> parameterList = getMethodUsageParameters(methodUsage);
+        List<String> exceptionList = getMethodUsageExceptions(methodUsage);
+        return (methodModifiers + " " + (typeParameterList.isEmpty() ? "" : "<" + String.join(", ", typeParameterList) + ">") +
+                " " + getTypeWithoutPackages(methodDeclaration.getReturnType()) +
+                " " + methodDeclaration.getName() +
+                "(" + String.join(", ", parameterList) + ")" +
+                (exceptionList.isEmpty() ? "" : " throws " + String.join(", ", exceptionList)))
+                .replaceAll(" +", " ").trim();
+    }
+
+    /**
+     * Gets the package of a given compilation unit.
+     *
+     * @param cu a compilation unit
+     * @return the package of the compilation unit
+     * @throws PackageDeclarationNotFoundException if the package declaration
+     * cannot be found
+     */
+    public static PackageDeclaration getPackageDeclarationFromCompilationUnit(
+            CompilationUnit cu
+    ) throws PackageDeclarationNotFoundException {
+        Optional<PackageDeclaration> jpPackage = cu.getPackageDeclaration();
+        if (jpPackage.isEmpty()) {
+            throw new PackageDeclarationNotFoundException(
+                    "The JavaParser package declaration of the compilation unit is empty"
+            );
         }
-        String methodReturnType = getTypeWithoutPackages(methodUsage.returnType());
-        String methodName = methodUsage.getName();
-        List<String> methodParameters = new ArrayList<>();
-        for (int i=0; i < methodUsage.getNoParams(); i++) {
-            methodParameters.add(getTypeWithoutPackages(methodUsage.getParamType(i)) + " arg" + i);
+        return jpPackage.get();
+    }
+
+    /**
+     * Returns the base element type of a resolved type {@link ResolvedType}.
+     * Recursively strips all array variables. For example:
+     *  Object[][] => Object
+     *
+     * @param resolvedType a type
+     * @return the base component type
```

Comment:
This uses "component" (which is correct), but the first sentence of the documentation uses "element".  Please be consistent.